src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
+        } else {
+            this.dataArray = new Object[capacity];
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfFull();
+        addOnNextFreeSell(value);
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexValidation(index);
+        if (index == size && size != dataArray.length) {
+            addOnNextFreeSell(value);
+        } else {
+            addByIndex(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+        if (list.size() + size > dataArray.length) {
+            growTo(dataArray, list);
+        }
+        for (int i = 0; i < list.size(); i++) {
 
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        if (isIndexExist(index)) {
+            return (T) dataArray[index];
+        }
         return null;
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (isIndexExist(index)) {
+            dataArray[index] = value;
+        }
     }
 
     @Override
     public T remove(int index) {
+        if (isIndexExist(index)) {
+            return removeElement(index);
+        }
         return null;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (areObjectsEquals(element, dataArray[i])) {
+                return removeElement(i);
+            }
+        }
+        throw new NoSuchElementException("There aren't such elements in the list like " + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void growIfFull() {
+        if (dataArray.length == size) {
+            grow(dataArray);
+        }
+    }
+
+    private void grow(Object[] fullArray) {
+        dataArray = new Object[fullArray.length + fullArray.length / 2];
+        System.arraycopy(fullArray, 0, dataArray, 0, fullArray.length);
+    }
+
+    private void growTo(Object[] fullArray, List<T> list) {
+        dataArray = new Object[size + list.size()];
+        System.arraycopy(fullArray, 0, dataArray, 0, fullArray.length);
+    }
+
+    private void addSize() {
+        size++;
+    }
+
+    private void addOnNextFreeSell(T value) {
+        growIfFull();
+        addSize();
+        dataArray[size - 1] = value;
+    }
+
+    private void addByIndex(T value, int index) {
+        growIfFull();
+        System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
+        dataArray[index] = value;
+        addSize();
+    }
+
+    private boolean isIndexExist(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("This index don't exist");
+        } else {
+            return true;
+        }
```

Comment:
overall I do not recommend to mix-up the boolean based validation and exception throwing validation, it's better to pick one and either make it void and name it like `checkIndex()` or make it return boolean and name it `isIndexValid()` and not throw and exception in it